Source/Algorithm/LAGraph_dense_relabel.c

```diff
@@ -0,0 +1,138 @@
+    See LICENSE file for more details.
+
+*/
+
+//------------------------------------------------------------------------------
+
+// LAGraph_dense_relabel: relabel sparse IDs to dense row/column indices
+// Contributed by Marton Elekes and Gabor Szarnyas,
+// Budapest University of Technology and Economics
+// (with accented characters: M\'{a}rton Elekes and G\'{a}bor Sz\'{a}rnyas.
+
+// Converts array of sparse IDs (ids) to row/column indices between 0...(nids-1).
+//
+// Gives back two binary matrices for conversion between ID- and index-based vertices.
+// id2index vector can be used to look up for indices of chosen IDs.
+// id_dimension gives back the height of Id2index matrix and id2index vector. (Same as width of Index2id_handle matrix.)
+//   id_dimension is the size that can store the largest ID in the array.
+//   Currently it is the largest valid dimension in SuiteSparse:GraphBLAS (GB_INDEX_MAX = 2^60)
+//
+// Find usage example in /Test/DenseRelabel/dense_relabel_test.c
+
+#include "LAGraph_internal.h"
+#include <string.h>
+
+#define LAGRAPH_FREE_ALL            \
+{                                   \
+    GrB_free (Id2index_handle) ;    \
+    GrB_free (Index2id_handle) ;    \
+    GrB_free (id2index_handle) ;    \
+    LAGRAPH_FREE (indices) ;        \
+    LAGRAPH_FREE (true_values) ;    \
+}
+
+//------------------------------------------------------------------------------
+
+GrB_Info LAGraph_dense_relabel               // relabel sparse IDs to dense row/column indices
+        (
+                GrB_Matrix *Id2index_handle, // output matrix: A(id, index)=1 (unfilled if NULL)
+                GrB_Matrix *Index2id_handle, // output matrix: B(index, id)=1 (unfilled if NULL)
+                GrB_Vector *id2index_handle, // output vector: v(id)=index (unfilled if NULL)
+                const GrB_Index *ids,        // array of unique identifiers (under GB_INDEX_MAX=2^60)
+                GrB_Index nids,              // number of identifiers
+                GrB_Index *id_dimension      // number of rows in Id2index matrix, id2index vector (unfilled if NULL)
+        ) {
+
+    GrB_Index *indices = NULL;
+    bool *true_values = NULL;
+
+    // from LAGraph_1_to_n.c
+    int nthreads = LAGraph_get_nthreads();
+    nthreads = LAGRAPH_MIN ((int64_t) (nids / 4096), nthreads);
+    nthreads = LAGRAPH_MAX (nthreads, 1);
+
+    //--------------------------------------------------------------------------
+    // check inputs
+    //--------------------------------------------------------------------------
+
+    if (!Id2index_handle && !Index2id_handle && !id2index_handle) {
+        LAGRAPH_ERROR ("All output mapping arguments are NULL", GrB_NULL_POINTER);
+    }
+    if (!ids) {
+        LAGRAPH_ERROR ("ids is NULL", GrB_NULL_POINTER);
+    }
+
+    // the largest valid dimension in SuiteSparse:GraphBLAS (GB_INDEX_MAX)
+    GrB_Index id_max_dimension = ((GrB_Index) (1ULL << 60));
```

Comment:
@DrTimothyAldenDavis
Is it possible to expose GB_INDEX_MAX macro on the user-facing API of SuiteSparse?

Reply:
Yes, that would be a good idea. I will do that in the next release.